### DISCUSSION TITLE
Writer的PreSql相关疑问

### DISCUSSION THREAD

**Discussion by me1dlinger** [wgzhao/Addax#687] [General]
执行带有presql的sqlwriter任务正常的逻辑应该是先reader出数据，再执行presql然后执行writer数据。现在似乎是同时presql和reader，如果reader时间很长并且presql是truncate表时，会导致表中无数据的时间比期待的长了很多。版本是4.0.9。

![image](https://user-images.githubusercontent.com/35567686/199152392-11f0c9b5-1772-4b23-b7f3-41fc1887d2bd.png)

**Comment by wgzhao** [comment]
reader 和 writer 的task 并不是串行的，所以的确会存在你说的这种情况
